### ISSUE TITLE
Discord Music Bot v5 - Dashboard

### ISSUE THREAD

**Issue by SudhanPlayz** [SudhanPlayz/Discord-MusicBot#638]
### Features

- [ ] Routes handler
- [ ] API Authorization 
- [ ] Discord Authorization 
- [ ] Socket.io Handler with Auth

### Endpoint

- [ ] POST: Add song to server
- [ ] GET: server queue
- [ ] GET: params(query) to search top 10 songs and return it
- [ ] GET: current song
- [ ] POST: Delete song from queue

### Socket.io

- [ ] Connection event
- [ ] Song added event
- [ ] Queue ended event
- [ ] Pause n play song event and handling from clients

### Web

- [ ] Design a beautiful panel UI

> *Note: This list in incomplete, In future maybe new things will be added*

**Comment by DarrenOfficial** [comment]
yeeeeeeeeeeeeeeeeeeeee

**Comment by JotaroKujo0525** [comment]
h

**Comment by SudhanPlayz** [comment]
Maybe I am free next month... if I am then I would do this

**Comment by DarrenOfficial** [comment]
> Maybe I am free next month... if I am then I would do this

Yay!!!!!!!!!!!! teach me react plz. I only know springboot

**Comment by SudhanPlayz** [comment]
> > Maybe I am free next month... if I am then I would do this
> 
> Yay!!!!!!!!!!!! teach me react plz. I only know springboot

Design in figma first then react

**Comment by k-g-n** [comment]
hmm

**Comment by DarrenOfficial** [comment]
Patiently waiting cause i don't know how to figma.

**Comment by SudhanPlayz** [comment]
> Patiently waiting cause i don't know how to figma.

Learn ;-;

**Comment by priyam1234-spec** [comment]
any eta on this?

**Comment by TheOnlyWayUp** [comment]
> [ ] POST: Delete song from queue

Convert to `[ ] DELETE: Delete song from queue`.

**Comment by DarrenOfficial** [comment]
No? theres no such thing as Delete

**Comment by DarrenOfficial** [comment]
Just an update, the v5 will use MySQL to store information such as but not limited to:
24/7 per server config
Music Queue (if bot restart it will picked up the previusly stored queue)
Dashboard related config
and maybe, just maybe, Playlist.

But Darren! Is mysql available on replit? yes
https://replit.com/talk/templates/Getting-started-with-MySql/33744 go

**Comment by TheOnlyWayUp** [comment]
> No? theres no such thing as Delete

https://reqbin.com/Article/HttpDelete

**Comment by DarrenOfficial** [comment]
> > No? theres no such thing as Delete
> 
> https://reqbin.com/Article/HttpDelete

This wouldn't work as what we're trying to accomplish is to allow a user from the dashboard to send a request through the API to delete something on the queue

It goes something like this
You → Dashboard(Sends a GET request for what's in the queue for X guild) → API (Sends all the songs in the queue for that "guilds") → You (get the list)
Let's say the  queue for that guild is
1. Him
2. Calling
3. Knee Deep At ATP

You → Dashboard(sends a POST request to delete song number 3, on guild X) → the song gets removed from the queue! everyone's happy yay

**Comment by TheOnlyWayUp** [comment]
Press a button -> Sends delete request

**Comment by DarrenOfficial** [comment]
no

**Comment by DarrenOfficial** [comment]
thats not how it works, you'll need to think it logically

**Comment by DarrenOfficial** [comment]
cause there's multiple variables, such as different guilds

**Comment by SudhanPlayz** [comment]
> > [ ] POST: Delete song from queue
> 
> Convert to `[ ] DELETE: Delete song from queue`.

i just wanna make everything POST and GET lol

**Comment by DarrenOfficial** [comment]
exactly, simple 🙏

**Comment by Anime-x** [comment]
DarkMan

**Comment by Vincenzo160** [comment]
What appended?

**Comment by popQA17** [comment]
Need help with react (nextjs)?

**Comment by DarrenOfficial** [comment]
if you can sure 👍🏼

**Comment by xAlcahest** [comment]
Any news about it?

**Comment by SudhanPlayz** [comment]
> Need help with react (nextjs)?

yup, you can help if you can. I will be pushing the nextjs project to v5 branch in some hours

**Comment by SudhanPlayz** [comment]
> Any news about it?

![](https://media.discordapp.net/attachments/855346697029812228/984083782451732540/unknown.png)

**Comment by XstreamSpeed** [comment]
> > Any news about it?
> 
> ![](https://camo.githubusercontent.com/9199bc3def813ae9e389f03c524209afe21d693f451f9552b0d9c3526895d7dd/68747470733a2f2f6d656469612e646973636f72646170702e6e65742f6174746163686d656e74732f3835353334363639373032393831323232382f3938343038333738323435313733323534302f756e6b6e6f776e2e706e67)

Yeah, and I have a few issues, not sure if its implemented or pending or what but when I click the login button, it says "Cannot GET /login" this continues to happen with /404, /dashboard, /index, /login, /logout and /servers. Is this a bug?

**Comment by XstreamSpeed** [comment]
> > > Any news about it?
> > 
> > 
> > ![](https://camo.githubusercontent.com/9199bc3def813ae9e389f03c524209afe21d693f451f9552b0d9c3526895d7dd/68747470733a2f2f6d656469612e646973636f72646170702e6e65742f6174746163686d656e74732f3835353334363639373032393831323232382f3938343038333738323435313733323534302f756e6b6e6f776e2e706e67)
> 
> Yeah, and I have a few issues, not sure if its implemented or pending or what but when I click the login button, it says "Cannot GET /login" this continues to happen with /404, /dashboard, /index, /login, /logout and /servers. Is this a bug?

For reference, the links looks like this `discord-music-bot.<user>.repl.co
`

**Comment by SudhanPlayz** [comment]
> > > Any news about it?
> > 
> > ![](https://camo.githubusercontent.com/9199bc3def813ae9e389f03c524209afe21d693f451f9552b0d9c3526895d7dd/68747470733a2f2f6d656469612e646973636f72646170702e6e65742f6174746163686d656e74732f3835353334363639373032393831323232382f3938343038333738323435313733323534302f756e6b6e6f776e2e706e67)
> 
> Yeah, and I have a few issues, not sure if its implemented or pending or what but when I click the login button, it says "Cannot GET /login" this continues to happen with /404, /dashboard, /index, /login, /logout and /servers. Is this a bug?

it's not a bug, it's the feature 😊
They will be working soon

**Comment by jchrobakk** [comment]
If you need frontend dev with react stack then you can hmu, I can help you guys with it but I'm just a junior, so I don't promise anything

**Comment by BossDaily** [comment]
Someone should come up with a rough concept in something like figma

**Comment by SudhanPlayz** [comment]
> Someone should come up with a rough concept in something like figma
> 

since I'm the only one who is coding the dashboard rn, I think its a waste for time now. I will just build it without using figma cuz all components are available within nextui and will be easy to code

**Comment by popQA17** [comment]
im not sure about using NextUI, its not really completed yet.

**Comment by popQA17** [comment]
Let me know if you need me to design the landing page :)

**Comment by SudhanPlayz** [comment]
> Let me know if you need me to design the landing page :)

If you can, then you can do it. Thanks for the contribution

**Comment by SoulHunter287** [comment]
ded

**Comment by SudhanPlayz** [comment]
> ded

not really

**Comment by IlyasMooerkens** [comment]
> > ded
> 
> not really

the api for v5 is not working is it still under dev?

**Comment by JotaroKujo0525** [comment]
> > > ded
> > 
> > 
> > not really
> 
> the api for v5 is not working is it still under dev?

ye

**Comment by popQA17** [comment]
Any news?

**Comment by Checker8763** [comment]
> Just an update, the v5 will use MySQL to store information such as but not limited to: 24/7 per server config Music Queue (if bot restart it will picked up the previusly stored queue) Dashboard related config and maybe, just maybe, Playlist.
> 
> But Darren! Is mysql available on replit? yes https://replit.com/talk/templates/Getting-started-with-MySql/33744 go

Why MySql and need an extra service to run?
I would have more thought of SQLite which has great performace even on medium trafic websites (100k/visit(day).
[claim](https://www.sqlite.org/whentouse.html#website)

the Workload of the Bot would be as far as I know just be writing and updating config occationaly
but reading a lot of data.

Thats a perfect [usecase for sqlite](https://www.sqlite.org/whentouse.html#dbcklst) 

And because it is a wide spread technology it has gread libs and the sql should stay the same or can even be enhancey because sqlite supports more complex querys usualy found on something like PostGres migrating the code should be realtivly easy.
Nomraly only the setup should be changed.

**Comment by jozef-drabik** [comment]
If there is something u need help with i would like to contribute. I am full-stack developer but mainly focus on next.js and react. HMU.

**Comment by thirsty-12** [comment]
any update?

**Comment by DarrenOfficial** [comment]
> @Checker8763 
> Why MySql and need an extra service to run? ... And ... only the setup should be changed.

To answer your question, MySQL because at the start this project was designed to run in hundreds of servers simultaneously. But since we all know what happened to big music bot… We now changed our focus for this bot to run on “small” server. Server with few members.
SQLite is valid. Will update the to-do list.


> @jozef-drabik 
> If there is something u need help with i would like to contribute. I am full-stack developer but mainly focus on next.js and react. HMU.

Any contribution is accepted and welcomed. Feel free to add features that you would like to see added.

> @thirsty-12 
> any update? @SudhanPlayz

AFAIK No, he has been pretty inactive in this project.

**Comment by DarrenOfficial** [comment]
To be completely transparent with you all, my schedule is very, very full. I only have free time on Sunday night. So some important issue on this GitHub *might* not get to me in time. Ahem, https://github.com/SudhanPlayz/Discord-MusicBot/issues/1156

**Comment by aVitomin** [comment]
project with dashboard probably died ? :D

**Comment by DarrenOfficial** [comment]
> project with dashboard probably died ? :D

On hold to be exact... 😄

**Comment by BLACK4585** [comment]
Hey,
what's the current status of v5? It already got mentioned in the latest release in 2021 "soon™". But it's still not out.
Thanks in advance.

**Comment by brianferri** [comment]
> Hey,
> what's the current status of v5? It already got mentioned in the latest release in 2021 "soon™". But it's still not out.
> Thanks in advance.

Working on it right now, but in v5.1. we time skipped.

Get live updates from
https://github.com/SudhanPlayz/Discord-MusicBot/pull/1227

**Comment by brianferri** [comment]
All this has been done in: https://github.com/wtfnotavailable/Discord-MusicBot

**Comment by fragtion** [comment]
@BioCla 
can you open issue requests on your repo? we need lavalink v4 support

**Comment by dpaes** [comment]
this still in develelopment?

**Comment by LewdHuTao** [comment]
> this still in develelopment?

![image](https://github.com/user-attachments/assets/e3fb0816-ea54-4f68-a647-ced562db0fdf)
